core/config/toml/types.go

```diff
@@ -149,8 +149,8 @@ func (d *DatabaseSecrets) ValidateConfig() (err error) {
 func (d *DatabaseSecrets) validateConfig(buildMode string) (err error) {
 	if d.URL == nil || (*url.URL)(d.URL).String() == "" {
 		err = multierr.Append(err, configutils.ErrEmpty{Name: "URL", Msg: "must be provided and non-empty"})
-	} else if *d.AllowSimplePasswords && buildMode == build.Prod {
-		err = multierr.Append(err, configutils.ErrInvalid{Name: "AllowSimplePasswords", Value: true, Msg: "insecure configs are not allowed on secure builds"})
+	} else if d.AllowSimplePasswords != nil && buildMode == build.Prod {
+		err = multierr.Append(err, configutils.ErrInvalid{Name: "AllowSimplePasswords", Value: "*", Msg: "AllowSimplePasswords is not allowed on secure builds"})
```

Comment:
Oh I see that you are checking `d.AllowSimplePasswords != nil`. This will never be true. The system is designed to fill in omitted fields with default values, which is why it was safe to dereference w/o a nil check before.